### PR TITLE
Cut out initialization of datetimes to `bitemporal_assign_initialize_value`

### PR DESCRIPTION
**Pull request by wata727**
Add a public method that just initializes valid_from/to and transaction_from/to of the new record.

This allows you to assign the values to instances without actually saving them, providing the possibility of updates outside of the activerecord-bitemporal interface.

### CODE REVIEW

**Comment by wata727** [comment]
We chatted internally to discuss the interface and make updates about it.

An important change is to make the `initialize_bitemporal` method private. This method is not intended to be exposed to gem users as a public API, but we may refer to this method for internal reasons in our application.

File: lib/activerecord-bitemporal/bitemporal.rb [hunk 1]
```diff
@@ -525,6 +510,25 @@ def reload(options = nil)
           self
         end
       end
+
+      private
+
+      def bitemporal_assign_initialize_value(valid_datetime, current_time)
```
Comment (osyo-manga) [inline lib/activerecord-bitemporal/bitemporal.rb:516]: Can you change to keyword arguments?
`bitemporal_assign_initialize_value(valid_datetime: Time.current, current_time: Time.current)` is more readable than `bitemporal_assign_initialize_value(Time.current, Time.current)`

Comment (wata727) [reply lib/activerecord-bitemporal/bitemporal.rb:516]: Sure, but was it intended that the default value for `valid_datetime` is `Time.current`? The current implementation is that the argument is required.

Comment (osyo-manga) [reply lib/activerecord-bitemporal/bitemporal.rb:516]: > The current implementation is that the argument is required.

Yes. Example implementation.

```ruby
def bitemporal_assign_initialize_value(valid_datetime:, current_time: Time.current)
  # ...
end
bitemporal_assign_initialize_value(valid_datetime: target_datetime, current_time: any_time_current)
```
Or maybe even `def bitemporal_assign_initialize_value(valid_datetime, current_time: Time.current)`.

Comment (wata727) [reply lib/activerecord-bitemporal/bitemporal.rb]: Fixed https://github.com/kufu/activerecord-bitemporal/pull/74/commits/32b4599b2b043cf7dd82db44247f0aa548ed8e79

Comment (osyo-manga) [reply lib/activerecord-bitemporal/bitemporal.rb]: Thanks! :)

**Review by osyo-manga — APPROVED** [review]
Good :+1:

**Review by f440 — APPROVED** [review]
👍

**Pull request merged**